frontend/src/lib/components/EditableText.svelte

```diff
@@ -0,0 +1,157 @@
+﻿<script lang="ts">
+	import { LoadingIcon } from '$lib/icons';
+	import { ZodString, z } from 'zod';
+	import { _changeProjectName } from '../../routes/project/[project_code]/+page';
+	import IconButton from './IconButton.svelte';
+	import { Form, lexSuperForm, lexSuperValidate } from '$lib/forms';
+
+	export let value: string | undefined | null = undefined;
+	export let disabled = false;
+	export let saveHandler: (newValue: string) => Promise<unknown>;
+	export let placeholder: string | undefined = undefined;
+	export let multiline = false;
+	export let validation: ZodString | undefined = undefined;
+
+	let initialValue: string | undefined | null;
+	let editing = false;
+	let saving = false;
+	const fieldId = 'id' + new Date().getTime();
```

Comment:
can we let the user of EditableText provide an id and just set a default value? like what we did in the Input component.